js_modules/dagit/packages/core/src/asset-graph/AssetNode.tsx

```diff
@@ -100,24 +99,51 @@ export const AssetNodeStatusBox: React.FC<{background: string}> = ({background,
   </Box>
 );
 
-export const AssetNodeStatusRow: React.FC<{
+interface StatusRowProps {
+  definition: AssetNodeFragment;
+  liveData: LiveDataForNode | undefined;
+}
+
+export const AssetNodeStatusRow: React.FC<StatusRowProps> = (props) => {
+  const info = buildAssetNodeStatusRow(props);
+  return <AssetNodeStatusBox background={info.background}>{info.content}</AssetNodeStatusBox>;
+};
+
+function getStepKey(definition: AssetNodeFragment) {
+  // Used for linking to the run with this step highlighted. We only support highlighting
+  // a single step, so just use the first one.
+  const firstOp = definition.opNames.length ? definition.opNames[0] : null;
+  return firstOp || '';
+}
+
+export function buildAssetNodeStatusRow({
+  definition,
+  liveData,
+}: {
```

Comment:
Might be worth putting together a quick test for this, at least for any of the more complex conditions.

Reply:
I'm a little torn about this -- there are storybooks for all of these cases and I feel like the errors are most likely going to be visual ones. Having tests eg: "expect it to say Materialized" would be a nice added layer but also give a false sense of security -- I wouldn't want someone to change this, see that the tests still pass, and merge it without actually looking at the storybooks for visual correctness...

I wonder if we could make some sort of PR review checkbox ala: "[] I have looked at the storybooks for this"?

Reply:
Update: I added tests for all the storybook cases. Just simple ones that verify the correct status text is displayed, but I think it'll keep bugs out of this.